src/lib/arch/aarch64/ExceptionHandler.cc

```diff
@@ -48,8 +48,8 @@ bool ExceptionHandler::init() {
         uint8_t outSize = static_cast<uint8_t>(out.size());
         stateChange = {ChangeType::REPLACEMENT, {R0}, {retval}};
         stateChange.memoryAddresses.push_back({argp, outSize});
-        stateChange.memoryAddressValues.push_back(
-            RegisterValue(reinterpret_cast<const char*>(out.data()), outSize));
+        stateChange.memoryAddressValues.push_back(RegisterValue(
+            reinterpret_cast<const uint8_t*>(out.data()), outSize));
```

Comment:
The `reinterpret_cast`ing throughout this function is alarming and requires further investigation

Reply:
Personally I think RegisterValue should have proper move semantics and you should be able to create a blank register of some size and move it into the vector